r2dbc-mysql/src/main/java/JaysncResult.kt

```diff
@@ -0,0 +1,23 @@
+package com.github.jasync.r2dbc.mysql
+
+import com.github.jasync.sql.db.ResultSet
+import com.github.jasync.sql.db.RowData
+import io.r2dbc.spi.Result
+import io.r2dbc.spi.RowMetadata
+import io.reactivex.Flowable
+import org.reactivestreams.Publisher
+import java.util.function.BiFunction
+
+
+class JaysncResult(private val resultSet: ResultSet) : Result {
+    private val metadata = JasyncMetadata(resultSet)
+
+    override fun getRowsUpdated(): Publisher<Int> {
+        return Flowable.just(resultSet.size)
+    }
+
+    override fun <T> map(mappingFunction: BiFunction<io.r2dbc.spi.Row, RowMetadata, out T>): Publisher<T> {
+        return Flowable.fromIterable<RowData>(resultSet)
```

Comment:
Do you read all rows at once and return afterwards a `ResultSet`? Or is `ResultSet` fetching data when it reaches the end of e.g. a cursor page?

Reply:
there is no inherent paging in the driver at the moment and all results are fetched before the future completes

Reply:
That's good to know and that is a good behavior already (could be an optimization in the future).

Some drivers (like Cassandra) have a `ResultSet` and as soon as the iterator reaches the end of a page, the iterator it goes into a blocking mode to fetch the next page. That's a nasty surprise because it's not obviously visible, and that was the background of my question.